File: test/showcase.py
"""Papyrus token showcase — Python. The violet proof-stamp lives on `self`/`cls`."""

from __future__ import annotations          # keyword(from/import) · module
import math                                 # module name
from dataclasses import dataclass, field    # module · names

MAX_DEPTH: int = 12                          # constant · type annotation · number
TAU = math.pi * 2                            # builtin module · operator · number
LABEL = "drafting ink"                       # string
_pattern = r"\$\{(\w+)\}"                     # raw string / regexp-ish escapes


@dataclass                                   # decorator
class Blueprint:                             # class name
    title: str                              # property · type
    scale: float = 1.0
    layers: list[str] = field(default_factory=list)

    def add(self, layer: str) -> "Blueprint":   # method · self (violet) · param
        self.layers.append(layer)               # self (violet) · property
        return self                             # self (violet)

    @classmethod                                # decorator
    def blank(cls) -> "Blueprint":              # cls (violet) · method
        return cls(title="untitled")            # cls (violet)

    @property                                   # decorator
    def is_empty(self) -> bool:                 # self (violet) · builtin type
        return len(self.layers) == 0            # builtin · operator


def render(items: list[Blueprint]) -> None:     # function · param · type
    for i, bp in enumerate(items):              # keyword.control · builtin
        ok = bp.scale > 0 and bp.title != ""    # operators · None/True/False below
        status = "ok" if ok else None           # language constants
        print(f"{i}: {bp.title} -> {status}")   # builtin · f-string expression


if __name__ == "__main__":                      # language const · operator
    render([Blueprint.blank()])

File: test/showcase.ts
// Papyrus token showcase — TypeScript. Open under each theme and eyeball every role.
// Roles annotated so a reviewer knows which token maps to which design-language color.

import { readFile } from "node:fs/promises";          // keyword(import) · module(node:fs)
import * as path from "node:path";                    // namespace

const MAX_RETRIES = 3;                                // constant · number
const GREETING = "blueprint ink";                     // string
const pattern = /^\$\{(\w+)\}$/g;                      // regexp · escapes

type Role = "keyword" | "string" | "number";          // type alias · string union
interface Drawing {                                    // interface name
  title: string;                                       // property · type annotation
  scale: number;
  layers: readonly string[];
}

enum Ink { Iron, Sepia, Olive }                        // enum (class-ish) + members

@sealed                                                // decorator
class DraftingTable<T extends Drawing> {               // class name · type param
  private sheets: T[] = [];                            // storage modifier · property

  constructor(public readonly name: string) {}         // parameter

  add(sheet: T): this {                                // method · self/this return
    this.sheets.push(sheet);                           // this
    return this;                                       // this
  }

  get count(): number {                                // accessor
    return this.sheets.length ?? 0;                    // operator(??) · property
  }
}

async function render(table: DraftingTable<Drawing>): Promise<void> {  // function · async
  for (const sheet of table.sheets) {                  // keyword.control
    const ok = sheet.scale > 1 && sheet.title !== "";  // operators · boolean cmp
    console.log(`${sheet.title}: ${ok ? "ok" : "skip"}`);  // template expression
  }
}

function sealed(ctor: Function) {                      // builtin-ish · param
  Object.seal(ctor);                                   // support.function
}

export { DraftingTable, render, MAX_RETRIES, GREETING, pattern };  // language const below
export const isReady = true;                           // constant.language.boolean
